src/tools/tidy/src/features.rs

```diff
@@ -175,6 +175,45 @@ pub fn check(
         tidy_error!(bad, "Found {} features without a gate test.", gate_untested.len());
     }
 
+    let (version, channel) = get_version_and_channel(src_path);
+
+    let all_features_iter = features
+        .iter()
+        .map(|feat| (feat, "lang"))
+        .chain(lib_features.iter().map(|feat| (feat, "lib")));
+    for ((feature_name, feature), kind) in all_features_iter {
+        let since = if let Some(since) = feature.since { since } else { continue };
+        if since > version {
+            tidy_error!(
+                bad,
+                "The stabilization version {} of {} feature `{}` is newer than the current {}",
+                since,
+                kind,
+                feature_name,
+                version
```

Comment:
do these not support fmt args capture?

```suggestion
                "The stabilization version {since} of {kind} feature `{feature_name}` is newer than the current {version}",
```

Reply:
The macro actually didn't support this use case (only format args capture, no arguments afterwards) and this made me believe it's not supported at all, but a simple change of the macro fixed that. I've added that to the commit, and it works now. Thanks!